KOTLIN_UPGRADE.md

```diff
@@ -0,0 +1,38 @@
+# Checklist for upgrading to the new Kotlin version
+
+1. Create a new branch based off `develop`:
+   ```sh
+   git fetch
+   git checkout -b kotlin-<version> origin/develop
+   ```
+
+1. In `gradle.properties`,
+   set `kotlin_version` to the `<version>`.
+
+2. Likewise, set `kotlin_version` in `integration-testing/gradle.properties`.
+
+3. In `README.md`,
+   replace the occurrences of the old version with the `<version>`.
+   These are:
+   - A badge at the top of the file.
+   - The Kotlin version corresponding to the latest release,
+     mentioned in the first paragraph.
+   - Maven and Gradle configuration blocks.
+
+4. In `README.md`, find the link to https://pl.kotl.in/ and open it.
+   Set the compiler version of Kotlin Playground to the same major version as `<version>`,
+   click "Copy link", and replace the original link in `README.md` with the new one.
+
+5. Perform a full build: `gradle clean check publishToMavenLocal`,
+   followed by `gradle clean check` in the `integration-testing` directory.
+   Fix any issues that arise.
```

Comment:
Shall we have a separate mini page / section about building the project locally?

1. I had to install tvOS and other simulators through xcode, and it was not a trivial "next, next, ok". (refer to our DMs)

2. I had to run it like this

`./integration-testing/gradlew clean check`

Rather than `cd integration-testing`, `./gradlew clean check`

I don't know if this is obvious gradle 101, but since we are documenting everything anyway, why not.

Reply:
1. I haven't had to do that for a long time, but if installing emulators is non-trivial the right place for mentioning this is https://kotlinlang.org/docs/native-get-started.html, I believe. Right now, it just says
> If you use a Mac and want to create and run applications for macOS or other Apple targets, you also need to install [Xcode Command Line Tools](https://developer.apple.com/download/), launch it, and accept the license terms first.
2. In that case, you didn't run the correct code. `./integration-testing/gradlew clean check` means "run the tasks `clean` and `check` in the project from the current directory, using the Gradle version from the `integration-testing` directory".

Reply:
2. ack

Reply:
How about we keep it as is right now, and I'll do a proper build page later?

This is mostly a mac issue, and I assume you don't really use mac. I was going to reinstall my mac anyway, I'll document everything I need to install to build kotlinx.coroutines locally on mac.

Reply:
It did require me to install the emulator, though.